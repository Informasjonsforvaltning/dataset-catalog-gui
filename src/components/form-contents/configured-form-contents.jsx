import { reduxForm } from 'redux-form';

import FormContentsComponent from './form-contents.component';
import validate from './form-contents-validations';
import { asyncValidateDatasetInvokePatch } from '../dataset-registration-form/formsLib/asyncValidateDatasetInvokePatch';

const config = {
  form: 'contents',
  validate,
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormContents = reduxForm(config)(FormContentsComponent);
