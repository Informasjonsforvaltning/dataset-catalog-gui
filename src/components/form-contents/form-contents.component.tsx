import React, { memo, FC } from 'react';
import { compose } from 'redux';
import { FieldArray } from 'redux-form';

import {
  withTranslations,
  Props as TranslationsProps
} from '../../providers/translations';

import Helptext from '../helptext/helptext.component';
import TextAreaField from '../fields/field-textarea/field-textarea.component';
import MultilingualField from '../multilingual-field/multilingual-field.component';
import Standard from './standard/standard.component';
import InputFieldReadonly from '../fields/field-input-readonly/field-input-readonly.component';

interface ExternalProps {
  languages: any[];
  isReadOnly: boolean;
}

interface Props extends ExternalProps, TranslationsProps {}

const FormContentsComponent: FC<Props> = ({
  languages,
  isReadOnly,
  translationsService
}) => (
  <form>
    <div className='form-group mb-0'>
      <div className='mt-4'>
        <div className='form-group mb-0'>
          <Helptext
            title={translationsService.translate(
              'schema.content.helptext.conformsTo'
            )}
            term='Dataset_conformsTo'
          />
          <FieldArray
            name='conformsTo'
            component={Standard}
            titleLabel={translationsService.translate(
              'schema.common.titleLabel'
            )}
            linkLabel={translationsService.translate('schema.common.linkLabel')}
            languages={languages}
            isReadOnly={isReadOnly}
          />
        </div>
      </div>
    </div>
    <div className='form-group mb-0'>
      <Helptext
        title={translationsService.translate(
          'schema.content.helptext.relevance'
        )}
        term='Dataset_hasQualityAnnotation_relevance'
      />
      <MultilingualField
        name='hasRelevanceAnnotation.hasBody'
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={translationsService.translate(
          'schema.content.hasRelevanceAnnotationLabel'
        )}
        languages={languages}
      />
    </div>
    <div className='form-group mb-0'>
      <Helptext
        title={translationsService.translate(
          'schema.content.helptext.completeness'
        )}
        term='Dataset_hasQualityAnnotation_completeness'
      />
      <MultilingualField
        name='hasCompletenessAnnotation.hasBody'
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={translationsService.translate(
          'schema.content.hasCompletenessAnnotationLabel'
        )}
        languages={languages}
      />
    </div>
    <div className='form-group mb-0'>
      <Helptext
        title={translationsService.translate(
          'schema.content.helptext.accuracy'
        )}
        term='Dataset_hasQualityAnnotation_accuracy'
      />
      <MultilingualField
        name='hasAccuracyAnnotation.hasBody'
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={translationsService.translate(
          'schema.content.hasAccuracyAnnotationLabel'
        )}
        languages={languages}
      />
    </div>
    <div className='form-group mb-0'>
      <Helptext
        title={translationsService.translate(
          'schema.content.helptext.availability'
        )}
        term='Dataset_hasQualityAnnotation_availability'
      />
      <MultilingualField
        name='hasAvailabilityAnnotation.hasBody'
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={translationsService.translate(
          'schema.content.hasAvailabilityAnnotationLabel'
        )}
        languages={languages}
      />
    </div>
  </form>
);

export default compose<FC<ExternalProps>>(
  memo,
  withTranslations
)(FormContentsComponent);
