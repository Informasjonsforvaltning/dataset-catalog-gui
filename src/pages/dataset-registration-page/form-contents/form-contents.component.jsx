import React from 'react';
import PropTypes from 'prop-types';
import { FieldArray } from 'redux-form';

import localization from '../../../services/localization';
import { Helptext } from '../../../components/helptext/helptext.component';
import TextAreaField from '../../../components/fields/field-textarea/field-textarea.component';
import MultilingualField from '../../../components/multilingual-field/multilingual-field.component';
import { Standard } from './standard/standard.component';
import InputFieldReadonly from '../../../components/fields/field-input-readonly/field-input-readonly.component';

export const FormContentsComponent = ({ languages, isReadOnly }) => (
  <form>
    <div className="form-group">
      <div className="mt-4">
        <div className="form-group">
          <Helptext
            title={localization.schema.content.helptext.conformsTo}
            term="Dataset_conformsTo"
          />
          <FieldArray
            name="conformsTo"
            component={Standard}
            titleLabel={localization.schema.common.titleLabel}
            linkLabel={localization.schema.common.linkLabel}
            languages={languages}
            isReadOnly={isReadOnly}
          />
        </div>
      </div>
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.content.helptext.relevance}
        term="Dataset_hasQualityAnnotation_relevance"
      />
      <MultilingualField
        name="hasRelevanceAnnotation.hasBody"
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={localization.schema.content.hasRelevanceAnnotationLabel}
        languages={languages}
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.content.helptext.completeness}
        term="Dataset_hasQualityAnnotation_completeness"
      />
      <MultilingualField
        name="hasCompletenessAnnotation.hasBody"
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={localization.schema.content.hasCompletenessAnnotationLabel}
        languages={languages}
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.content.helptext.accuracy}
        term="Dataset_hasQualityAnnotation_accuracy"
      />
      <MultilingualField
        name="hasAccuracyAnnotation.hasBody"
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={localization.schema.content.hasAccuracyAnnotationLabel}
        languages={languages}
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.content.helptext.availability}
        term="Dataset_hasQualityAnnotation_availability"
      />
      <MultilingualField
        name="hasAvailabilityAnnotation.hasBody"
        component={isReadOnly ? InputFieldReadonly : TextAreaField}
        label={localization.schema.content.hasAvailabilityAnnotationLabel}
        languages={languages}
      />
    </div>
  </form>
);

FormContentsComponent.defaultProps = {
  languages: [],
  isReadOnly: false
};

FormContentsComponent.propTypes = {
  languages: PropTypes.array,
  isReadOnly: PropTypes.bool
};
