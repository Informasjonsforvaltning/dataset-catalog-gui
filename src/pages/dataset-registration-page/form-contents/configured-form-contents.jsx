import { reduxForm } from 'redux-form';

import { FormContentsComponent } from './form-contents.component';
import validate from './form-contents-validations';
import { asyncValidateDatasetInvokePatch } from '../formsLib/asyncValidateDatasetInvokePatch';

const config = {
  form: 'contents',
  validate,
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormTitle = reduxForm(config)(FormContentsComponent);
