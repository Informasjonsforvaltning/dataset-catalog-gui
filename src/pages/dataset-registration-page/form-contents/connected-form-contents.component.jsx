import { getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';

import { ConfiguredFormTitle } from './configured-form-contents';
import {
  conformsToType,
  relevanceAnnotationType,
  completenessAnnotationType,
  accuracyAnnotationType,
  availabilityAnnotationType
} from '../../../schemaTypes';

const mapStateToProps = ({ state }, { datasetItem = {} }) => {
  const {
    conformsTo = [conformsToType],
    hasRelevanceAnnotation = relevanceAnnotationType,
    hasCompletenessAnnotation = completenessAnnotationType,
    hasAccuracyAnnotation = accuracyAnnotationType,
    hasAvailabilityAnnotation = availabilityAnnotationType
  } = datasetItem;

  return {
    initialValues: {
      conformsTo,
      hasRelevanceAnnotation,
      hasCompletenessAnnotation,
      hasAccuracyAnnotation,
      hasAvailabilityAnnotation
    },
    syncErrors: getFormSyncErrors('contents')(state)
  };
};

export const ConnectedFormContents = connect(mapStateToProps)(
  ConfiguredFormTitle
);
