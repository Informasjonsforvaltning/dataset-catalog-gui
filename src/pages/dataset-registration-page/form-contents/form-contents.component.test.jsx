import { shallow } from 'enzyme';
import { FormContentsComponent } from './form-contents.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {};
  wrapper = shallow(FormContentsComponent(defaultProps));
});

test('should render FormContentsComponent, { renderS correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
