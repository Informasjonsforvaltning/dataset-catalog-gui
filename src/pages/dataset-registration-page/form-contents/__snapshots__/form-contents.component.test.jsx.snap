// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormContentsComponent, { renderS correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <div
      className="mt-4"
    >
      <div
        className="form-group"
      >
        <withStateHandlers(Component)
          term="Dataset_conformsTo"
          title="Standard"
        />
        <FieldArray
          component={[Function]}
          linkLabel="Lenke"
          name="conformsTo"
          titleLabel="Tittel"
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="Dataset_hasQualityAnnotation_relevance"
      title="Relevans"
    />
    <MultilingualField
      component={[Function]}
      label="Relevans"
      languages={Array []}
      name="hasRelevanceAnnotation.hasBody"
      showLabel={false}
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="Dataset_hasQualityAnnotation_completeness"
      title="Kompletthet"
    />
    <MultilingualField
      component={[Function]}
      label="Kompletthet"
      languages={Array []}
      name="hasCompletenessAnnotation.hasBody"
      showLabel={false}
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="Dataset_hasQualityAnnotation_accuracy"
      title="Nøyaktighet"
    />
    <MultilingualField
      component={[Function]}
      label="Nøyaktighet"
      languages={Array []}
      name="hasAccuracyAnnotation.hasBody"
      showLabel={false}
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="Dataset_hasQualityAnnotation_availability"
      title="Tilgjengelighet"
    />
    <MultilingualField
      component={[Function]}
      label="Tilgjengelighet"
      languages={Array []}
      name="hasAvailabilityAnnotation.hasBody"
      showLabel={false}
    />
  </div>
</form>
`;
